src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -634,26 +634,16 @@ public static void Encrypt(string path)
         {
             if (path == null)
                 throw new ArgumentNullException(nameof(path));
```

Comment:
I don't like that it can throw an argument exception before one reaches PNSE... but likely not worth changing it.

Reply:
@JeremyKuhne @pjanotti let me know if i need to amend!

Reply:
Yes, please :)

Reply:
Ops, lost the race with the merge, no problem. You can leave as it is.

Reply:
Yes, please :)

Reply:
@pjanotti i'm working on part 2 of PR but if i align repo i cannot compile anymore on my win...seems related to this PR:
```
F:\git\corefx\pkg\frameworkPackage.targets(124,5): error : Cycle detected for F:\git\corefx\bin\Windows_NT.AnyCPU.Debug\System.IO.FileSystem\netcoreapp\System.IO.FileSystem.dll. [F:\git\corefx\pkg\Microsoft.Private.CoreFx.NETCoreApp\Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
F:\git\corefx\pkg\frameworkPackage.targets(124,5): error : System.IO.FileSystem > System.IO.FileSystem.DriveInfo > System.IO.FileSystem [F:\git\corefx\pkg\Microsoft.Private.CoreFx.NETCoreApp\Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
```
Is this only my local issue?

Reply:
> I don't like that it can throw an argument exception before one reaches PNSE

We've actually standardized on that as being desired, as it keeps as much behavior and validation consolidated and the same across platforms, and if a platform is then ever able to start supporting the functionality, you don't start going from a PNSE to an ANE.